pkg/client/client.go

```diff
@@ -26,3 +26,133 @@ type Client interface {
 	// MapParams mapping param, uri, query, body ...
 	MapParams(req *Request) (reqData interface{}, err error)
 }
+
+// MapOption option map, key : name, value : option
+type MapOption map[string]IOption
+
+var DefaultMapOption = MapOption{
+	"types":       &dubboParamTypesOpt{},
+	"group":       &dubboGroupOpt{},
+	"version":     &dubboVersionOpt{},
+	"interface":   &dubboInterfaceOpt{},
+	"application": &dubboApplicationOpt{},
+	"method":      &dubboMethodOpt{},
+}
+
+// IOption option interface.
+type IOption interface {
+	// Usable if option can use
+	Usable() bool
+	// SetUsable set usable
+	SetUsable(b bool)
+	// Action do with val for special
+	Action(req *Request, val interface{})
+}
+
+// CommonOption
+type CommonOption struct {
+	usable bool
+	IOption
+}
+
+// Usable get usable
+func (opt *CommonOption) Usable() bool {
+	return opt.usable
+}
+
+// SetUsable set usable
+func (opt *CommonOption) SetUsable(b bool) {
+	opt.usable = b
+}
+
+type dubboParamTypesOpt struct {
```

Comment:
should we place these client-specific options back to the dubbo package?

Reply:
I think so. But I not sure http is need this.